File: module/visual-config/src/main/java/org/visual/config/ConfigModule.java
package org.visual.config;

import com.google.inject.AbstractModule;
import lombok.val;
import org.github.gestalt.config.guice.GestaltModule;
import org.visual.config.provider.ConfigProvider;

public class ConfigModule extends AbstractModule {
  @Override
  protected void configure() {
    val gestalt = new ConfigProvider().get();
    install(new GestaltModule(gestalt));
  }
}

File: visual-model-app/src/main/kotlin/org/visual/model/app/module/ConfigModule.kt
package org.visual.model.app.module

import com.google.inject.AbstractModule
import org.github.gestalt.config.Gestalt
import org.github.gestalt.config.kotlin.getConfig
import org.visual.model.app.config.UI


class ConfigModule(
    private val gestalt: Gestalt
) : AbstractModule() {

    override fun configure() {
        val ui: UI = gestalt.getConfig("ui")
        System.err.println(ui)
    }
}